syntax = "proto3";
package huzch;
import "base.proto";
option cc_generic_services = true;

//用户名注册   
message UserRegisterReq {
    string request_id = 1;
    string user_name = 2;
    string password = 3;
}
message UserRegisterRsp {
    string request_id = 1;
    bool success = 2;
    optional string errmsg = 3;
}

//用户名登录 
message UserLoginReq {
    string request_id = 1;
    string user_name = 2;
    string password = 3;
}
message UserLoginRsp {
    string request_id = 1;
    bool success = 2;
    optional string errmsg = 3;
    optional string login_session_id = 4;
}

//手机号验证码获取
message GetPhoneVerifyCodeReq {
    string request_id = 1;
    string phone_number = 2;
}
message GetPhoneVerifyCodeRsp {
    string request_id = 1;
    bool success = 2;
    string errmsg = 3;
    string verify_code_id = 4;
}

//手机号注册
message PhoneRegisterReq {
    string request_id = 1;
    string phone_number = 2;
    string verify_code_id = 3;
    string verify_code = 4;
}
message PhoneRegisterRsp {
    string request_id = 1;
    bool success = 2;
    string errmsg = 3;
}

//手机号登录
message PhoneLoginReq {
    string request_id = 1;
    string phone_number = 2;
    string verify_code_id = 3;
    string verify_code = 4;
}
message PhoneLoginRsp {
    string request_id = 1;
    bool success = 2;
    string errmsg = 3; 
    string login_session_id = 4;
}

message GetUserInfoReq {
    string request_id = 1;
    optional string user_id = 2;
    optional string session_id = 3;
}
message GetUserInfoRsp {
    string request_id = 1;
    bool success = 2;
    string errmsg = 3; 
    UserInfo user_info = 4;
}

message GetMultiUserInfoReq {
    string request_id = 1;
    repeated string users_id = 2;
}
message GetMultiUserInfoRsp {
    string request_id = 1;
    bool success = 2;
    string errmsg = 3; 
    map<string, UserInfo> users_info = 4;
}

//用户头像修改 
message SetUserAvatarReq {
    string request_id = 1;
    optional string user_id = 2;
    optional string login_session_id = 3;
    bytes avatar = 4;
}
message SetUserAvatarRsp {
    string request_id = 1;
    bool success = 2;
    string errmsg = 3; 
}

//用户昵称修改 
message SetUserNameReq {
    string request_id = 1;
    optional string user_id = 2;
    optional string login_session_id = 3;
    string user_name = 4;
}
message SetUserNameRsp {
    string request_id = 1;
    bool success = 2;
    string errmsg = 3; 
}

//用户签名修改 
message SetUserDescriptionReq {
    string request_id = 1;
    optional string user_id = 2;
    optional string login_session_id = 3;
    string description = 4;
}
message SetUserDescriptionRsp {
    string request_id = 1;
    bool success = 2;
    string errmsg = 3; 
}

//用户手机修改 
message SetUserPhoneNumberReq {
    string request_id = 1;
    optional string user_id = 2;
    optional string login_session_id = 3;
    string phone_number = 4;
    string phone_verify_code_id = 5;
    string phone_verify_code = 6;
}
message SetUserPhoneNumberRsp {
    string request_id = 1;
    bool success = 2;
    string errmsg = 3; 
}

service UserService {
    rpc UserRegister(UserRegisterReq) returns (UserRegisterRsp);
    rpc UserLogin(UserLoginReq) returns (UserLoginRsp);
    rpc GetPhoneVerifyCode(GetPhoneVerifyCodeReq) returns (GetPhoneVerifyCodeRsp);
    rpc PhoneRegister(PhoneRegisterReq) returns (PhoneRegisterRsp);
    rpc PhoneLogin(PhoneLoginReq) returns (PhoneLoginRsp);
    rpc GetUserInfo(GetUserInfoReq) returns (GetUserInfoRsp);
    rpc GetMultiUserInfo(GetMultiUserInfoReq) returns (GetMultiUserInfoRsp);
    rpc SetUserAvatar(SetUserAvatarReq) returns (SetUserAvatarRsp);
    rpc SetUserName(SetUserNameReq) returns (SetUserNameRsp);
    rpc SetUserDescription(SetUserDescriptionReq) returns (SetUserDescriptionRsp);
    rpc SetUserPhoneNumber(SetUserPhoneNumberReq) returns (SetUserPhoneNumberRsp);
}
